src/tests/system/tests/test_gpo.py

```diff
@@ -0,0 +1,1056 @@
+"""
+SSSD GPO tests
+
+:requirement: IDM-SSSD-REQ: Group policy host based access control
+
+The following features and parameters are not covered:
+- ad_gpo_cache_timeout
+- ad_gpo_map_network
+- ad_gpo_map_batch
+- ad_gpo_map_service
+- ad_gpo_map_permit
+- ad_gpo_map_deny
+"""
+
+from __future__ import annotations
+
+import pytest
+
+from sssd_test_framework.roles.ad import AD
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.topology import KnownTopology
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.parametrize("method", ["su", "ssh"])
+@pytest.mark.topology(KnownTopology.AD)
+def test_gpo__is_set_to_enforcing_and_users_are_allowed(client: Client, ad: AD, method: str):
+    """
+    :title: Group policy object host base access control is set to enforcing and users are allowed
+    :description:
+        When ad_gpo_access_control is set to enforcing, user access is managed and enforced by the policy.
+        Unlisted users are denied access.
+    :setup:
+        1. Create the following users; 'user1', 'user2'
+        2. Create the following groups; 'group' with member 'user2'
+        3. Create and link the GPO 'site policy' and add 'user1', group and 'Domain Admins' to
+           SeInteractiveLogonRight key.
+        4. Configure sssd.conf with 'ad_gpo_access_control' = 'enforcing'
+        5. Start SSSD
+    :steps:
+        1. Authenticate as 'user1' and 'user2'
+    :expectedresults:
+        1. Users authentication is successful
+    :customerscenario: True
+    """
+    user1 = ad.user("user1").add()
+    user2 = ad.user("user2").add()
+    group = ad.group("group").add().add_members([user2])
+
+    ad.gpo("site policy").add().policy(
+        {
+            "SeInteractiveLogonRight": [user1, group, ad.group("Domain Admins")],
+            "SeDenyInteractiveLogonRight": [],
+        }
+    ).link()
+
+    client.sssd.domain["ad_gpo_access_control"] = "enforcing"
+    client.sssd.start()
+
+    assert client.auth.parametrize(method).password(username="user1", password="Secret123")
+    assert client.auth.parametrize(method).password(username="user2", password="Secret123")
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.parametrize("method", ["su", "ssh"])
+@pytest.mark.topology(KnownTopology.AD)
+def test_gpo__is_set_to_enforcing_and_users_are_denied(client: Client, ad: AD, method: str):
+    """
+    :title: Group policy object host base access control is set to enforcing and users are denied
+    :description:
+        When ad_gpo_access_control is set to enforcing, user access is managed and enforced by the policy.
+        Unlisted users are denied access.
+    :setup:
+        1. Create the following users; 'user1', 'deny_user1', 'deny_user2'
+        2. Create the 'deny_group' and add 'deny_user2' as a member
+        3. Create and link the GPO 'site policy' and add 'Domain Admins' to SeInteractiveLogonRight
```

Comment:
Same as above, why is `Domain Admins` needed?

You list `user1` but it is currently not used in the test. I would like to suggest to add `Domain Users` to `SeInteractiveLogonRight` to allow access for all users, so `user1` should be able to log in. With this it should be more clear that only users covered by `SeDenyInteractiveLogonRight` cannot log in while all others can.

Reply:
'user1' in the docs is a typo, it should not have be there.